src/Stack/Config.hs

```diff
@@ -157,16 +157,19 @@ makeConcreteResolver ar = do
                 iopc <- loadConfigYaml (parseProjectAndConfigMonoid (parent fp)) fp
                 ProjectAndConfigMonoid project _ <- liftIO iopc
                 return $ projectResolver project
-            ARLatestNightly -> return $ nightlySnapshotLocation $ snapshotsNightly snapshots
-            ARLatestLTSMajor x ->
+            ARLatestNightly -> nightlySnapshotLocation . snapshotsNightly <$> getSnapshots
+            ARLatestLTSMajor x -> do
+                snapshots <- getSnapshots
                 case IntMap.lookup x $ snapshotsLts snapshots of
                     Nothing -> throwString $ "No LTS release found with major version " ++ show x
                     Just y -> return $ ltsSnapshotLocation x y
-            ARLatestLTS
-                | IntMap.null $ snapshotsLts snapshots -> throwString "No LTS releases found"
-                | otherwise ->
-                    let (x, y) = IntMap.findMax $ snapshotsLts snapshots
-                     in return $ ltsSnapshotLocation x y
+            ARLatestLTS -> do
+                snapshots <- getSnapshots
+                let r | IntMap.null $ snapshotsLts snapshots = throwString "No LTS releases found"
```

Comment:
Minor nitpicking but will it not look simpler if `if` were used here? Then `let` will not be required and that trailing `r` as well...